### ISSUE TITLE
LIBERAÇÃO DE CÓDIGO-FONTE

### ISSUE THREAD

**Issue by pagoto** [gilmagno/spu-alfresco-data-dictionary#1]
Olá, assim como o SPDO do INTERLEGIS.
Vocês poderiam disponibilizar todo o Código-Fonte já instalavel, pois temos interesse em instalar em outros orgãos, e seria de grande valia.

**Comment by gilmagno** [comment]
Olá, Dhiego

O que tu está procurando? Um sistema de protocolo único?

Abraço

**Comment by pagoto** [comment]
Olá Gil Magno, sim! estamos procurando um Sistema de Protocolo Único!
Achamos interessante que o de vocês são baseado no Alfresco, sendo assim já
faria o trabalho de GED e PROTOCOLO ÚNICO.

2015-04-04 12:26 GMT-04:00 gilmagno notifications@github.com:

**Comment by gilmagno** [comment]
Opa, Dhiego

Então, eu não trabalho mais na prefeitura de Fortaleza, faz alguns anos. Desenvolvemos lá um sistema de protocolo único (SPU), e era mesmo interessante, por causa do Alfresco. No entanto, também deu muito trabalho quando estávamos fazendo, porque estávamos aprendendo e fazendo ao mesmo tempo.

O sistema mesmo, as regras, estava implementadas nos scripts do Alfresco. A interface web, que foi feita em PHP e Zend, apenas consumia os serviços que havíamos implementado no Alfresco.

O código mais complicado do sistema está no repositório que já está no github (estamos comentando nele hehe). O código em PHP eu não sei se ainda acho em algum backup por aqui, mas procuro e se achar te envio. :)

**Comment by gilmagno** [comment]
Há alguém na tua equipe que conheça mais ou menos o Alfresco?

**Comment by pagoto** [comment]
Olá Gil, há sim.. Inclusive ele esta de férias no pará!
Se acha-se o código PHP ajudaria muito, pois temos uma certa urgência nos
testes do serviço, estamos procurando implementar aqui na prefeitura de
Manaus e em parceria com o Banco do Povo!

Ajudaria 90% se conseguisse achar o código!

Se houver algo que eu e minha equipe possa ajudar também, estamos a
disposição amigo!

Desde já agradeço muito!!
Em 07/04/2015 21:22, "Gil Magno" notifications@github.com escreveu:

**Comment by gilmagno** [comment]
Certo, procurarei. De qualquer forma, a maneira que usávamos o PHP era, com o suporte a curl do PHP, nós acessávamos os serviços que estavam implementados no Alfresco. Então, por exemplo, quando na interface web preenchíamos um formulários com os dados de um processo e submetíamos o formulário, o que acontecia era um POST para o servidor do Alfresco (exemplo: ip:porta/alfresco/service/spu/processo/abrir), levando no seu "body" os dados do formulário. O serviço (implementado no Alfresco) recebia o POST, abria o processo e respondia para a interface web. A arquitetura era essa.

Inclusive, o código que usa o curl está aqui https://github.com/gilmagno/alfresco-php-api .

Então, qualquer coisa, te dou um alô!

É um prazer poder ajudar.
Um abraço!

**Comment by lucianoes** [comment]
Achei sensacional @gilmagno como foi feito o SPU usando o Alfresco. 

@pagoto conseguiu evoluir nesse desenvolvimento? Tenho procurado algo similar e talvez possa me dar algumas dicas ou unirmos esforços.
